polaris-react/src/components/Grid/Grid.tsx

```diff
@@ -32,17 +32,23 @@ export interface GridProps {
   columns?: Columns;
   children?: React.ReactNode;
 }
+
+const tokenToVariable = (value: SpacingTokens) => `var(--p-${value})`;
+
+const getSpacingValue = (value: SpacingTokens | undefined) =>
+  value ? tokenToVariable(value) : undefined;
+
 /** **Experimental!** This component is in alpha. Use with caution. */
 export const Grid: React.FunctionComponent<GridProps> & {
   Cell: typeof Cell;
 } = function Grid({gap, areas, children, columns}: GridProps) {
   const [gridTemplateAreas, setGridTemplateAreas] = useState(getAreas(areas));
   const style = {
-    '--pc-grid-gap-xs': gap?.xs,
-    '--pc-grid-gap-sm': gap?.sm,
-    '--pc-grid-gap-md': gap?.md,
-    '--pc-grid-gap-lg': gap?.lg,
-    '--pc-grid-gap-xl': gap?.xl,
+    '--pc-grid-gap-xs': getSpacingValue(gap?.xs),
+    '--pc-grid-gap-sm': getSpacingValue(gap?.sm),
+    '--pc-grid-gap-md': getSpacingValue(gap?.md),
+    '--pc-grid-gap-lg': getSpacingValue(gap?.lg),
+    '--pc-grid-gap-xl': getSpacingValue(gap?.xl),
```

Comment:
```suggestion
    '--pc-grid-gap-xs': gap?.xs && `var(${createVar(gap.xs)})`,
    '--pc-grid-gap-sm': gap?.sm && `var(${createVar(gap.sm)})`,
    '--pc-grid-gap-md': gap?.md && `var(${createVar(gap.md)})`,
    '--pc-grid-gap-lg': gap?.lg && `var(${createVar(gap.lg)})`,
    '--pc-grid-gap-xl': gap?.xl && `var(${createVar(gap.xl)})`,
```

Not sure we need the abstraction. Thoughts?